File: core/java/src/main/java/hara/lang/protocol/IPopLast.java
package hara.lang.protocol;

import hara.lang.declaration.HaraMethod;
import hara.lang.declaration.HaraProtocolBinding;

@HaraProtocolBinding(namespace = "std.protocol.ipoplast", name = "IPopLast")
public interface IPopLast {
  @HaraMethod(value = "pop-last", arity = 1)
  IPopLast popLast();
}

File: core/rust/src/lang/protocol/ipoplast.rs
use hara_protocol_macros::hara_protocol;

#[hara_protocol(namespace = "std.protocol.ipoplast", name = "IPopLast")]
pub trait IPopLast {
    type Output;

    #[hara_method(value = "pop-last", arity = 1)]
    fn pop_last(&self) -> Self::Output;
}
